File: app/schemas/customer_service.py
from datetime import datetime

from pydantic import BaseModel, Field, model_validator


class CustomerServiceConversationCreateRequest(BaseModel):
    target_type: str = Field(default="merchant", pattern="^(merchant|platform)$")
    merchant_id: int | None = Field(default=None, ge=1)
    product_id: int | None = Field(default=None, ge=1)
    order_id: int | None = Field(default=None, ge=1)
    initial_message: str | None = Field(default=None, max_length=1000)

    @model_validator(mode="after")
    def validate_target(self):
        if self.target_type == "merchant" and self.merchant_id is None:
            raise ValueError("商家客服会话必须指定 merchant_id")
        if self.target_type == "platform":
            self.merchant_id = None
        return self


class CustomerServiceMessageCreateRequest(BaseModel):
    content_type: str = Field(default="text", pattern="^(text|image)$")
    content: str = Field(default="", max_length=2000)
    image_urls: list[str] = Field(default_factory=list)

    @model_validator(mode="after")
    def validate_content(self):
        if not self.content.strip() and not self.image_urls:
            raise ValueError("消息内容不能为空")
        return self


class CustomerServiceConversationResponse(BaseModel):
    id: int
    user_id: int
    user_nickname: str | None = None
    target_type: str
    merchant_id: int | None = None
    merchant_name: str | None = None
    product_id: int | None = None
    product_name: str | None = None
    order_id: int | None = None
    order_no: str | None = None
    status: str
    last_message_at: datetime | None = None
    last_message: str | None = None
    unread_count: int = 0
    created_at: datetime
    updated_at: datetime


class CustomerServiceMessageResponse(BaseModel):
    id: int
    conversation_id: int
    sender_type: str
    sender_id: int
    sender_name: str | None = None
    content_type: str
    content: str
    image_urls: list[str]
    is_read: bool
    created_at: datetime

File: user-frontend/src/services/customerService.ts
import { http } from './http'

export type CustomerServiceConversation = {
  id: number
  user_id: number
  user_nickname?: string | null
  target_type: 'merchant' | 'platform'
  merchant_id?: number | null
  merchant_name?: string | null
  product_id?: number | null
  product_name?: string | null
  order_id?: number | null
  order_no?: string | null
  status: string
  last_message_at?: string | null
  last_message?: string | null
  unread_count: number
  created_at: string
  updated_at: string
}

export type CustomerServiceMessage = {
  id: number
  conversation_id: number
  sender_type: 'user' | 'merchant' | 'platform'
  sender_id: number
  sender_name?: string | null
  content_type: 'text' | 'image'
  content: string
  image_urls: string[]
  is_read: boolean
  created_at: string
}

export type PageResult<T> = {
  list: T[]
  page: number
  page_size: number
  total: number
}

export const customerService = {
  createConversation(payload: {
    target_type: 'merchant' | 'platform'
    merchant_id?: number | null
    product_id?: number | null
    order_id?: number | null
    initial_message?: string | null
  }) {
    return http.post<unknown, { data: CustomerServiceConversation }>('/customer-service/conversations', payload)
  },

  listConversations(params?: { status?: string; page?: number; page_size?: number }) {
    return http.get<unknown, { data: PageResult<CustomerServiceConversation> }>('/customer-service/conversations', { params })
  },

  listMessages(conversationId: number, params?: { page?: number; page_size?: number }) {
    return http.get<unknown, { data: PageResult<CustomerServiceMessage> }>(`/customer-service/conversations/${conversationId}/messages`, { params })
  },

  sendMessage(conversationId: number, payload: { content: string; content_type?: 'text' | 'image'; image_urls?: string[] }) {
    return http.post<unknown, { data: CustomerServiceMessage }>(`/customer-service/conversations/${conversationId}/messages`, {
      content_type: payload.content_type ?? 'text',
      content: payload.content,
      image_urls: payload.image_urls ?? [],
    })
  },
}
